VTEX - Orders API.json

```diff
@@ -5712,7 +5712,3709 @@
+            "post": {
+              "tags": [
+                "Change order"
+              ],
+              "summary": "Retry order change",
+              "operationId": "RetryChange",
+              "description": "This endpoint makes a new attempt to perform an order change previously created using the Order change endpoint that, for some reason, was not completely processed.\r\n\r\nTo make this request, you need the order's `changeRequestId`, its unique change identification code. When you retry to change an order using its `changeRequestId`, instead of creating a new change, you avoid duplications.\r\n\r\nThis endpoint's successful response status code is `204 No Content`, therefore with no response body.\r\n\r\n## Permissions\r\n\r\nAny user or [application key](https:\/\/developers.vtex.com\/docs\/guides\/authentication-overview#application-keys) must have at least one of the appropriate [License Manager resources](https:\/\/help.vtex.com\/en\/tutorial\/license-manager-resources--3q6ztrC8YynQf6rdc6euk3) to be able to successfully run this request. Otherwise they will receive a status code `403` error. These are the applicable resources for this endpoint:\r\n\r\n| **Product** | **Category** | **Resource** |\r\n| --------------- | ----------------- | ----------------- |\r\n| Checkout | CheckoutResources | **Orders Full Access** |\r\n| OMS | OMS access | **Cancel order** |\r\n| OMS | OMS access | **Change order** |\r\n\r\nYou can [create a custom role](https:\/\/help.vtex.com\/en\/tutorial\/roles--7HKK5Uau2H6wxE1rH5oRbc#creating-a-role) with that resource or use one of the following [predefined roles](https:\/\/help.vtex.com\/en\/tutorial\/predefined-roles--jGDurZKJHvHJS13LnO7Dy):\r\n\r\n| **Role** | **Resource** | \r\n| --------------- | ----------------- | \r\n| Checkout Admin | Orders Full Access |\r\n| OMS - Full access | <p>Cancel order<\/p><p>Change order<\/p> |\r\n| IntegrationProfile - Fulfillment Oms | <p>Cancel order<\/p><p>Change order<\/p> |\r\n| IntegrationProfile - Fulfillment Gateway | Cancel order |\r\n| IntegrationProfile - Fulfillment Gateway Oms | <p>Cancel order<\/p><p>Change order<\/p> |\r\n| inStore Sales Person | <p>Cancel order<\/p><p>Change order<\/p> |\r\n\r\n>\u2757 Assigning a [predefined role](https:\/\/help.vtex.com\/en\/tutorial\/predefined-roles--jGDurZKJHvHJS13LnO7Dy) to users or application keys usually grants permission to multiple [License Manager resources](https:\/\/help.vtex.com\/en\/tutorial\/license-manager-resources--3q6ztrC8YynQf6rdc6euk3). If some of these permissions are not necessary, consider creating a custom role instead. For more information regarding security, see [Best practices for using application keys](https:\/\/help.vtex.com\/en\/tutorial\/best-practices-application-keys--7b6nD1VMHa49aI5brlOvJm).\r\n\r\nTo learn more about machine authentication at VTEX, see [Authentication overview](https:\/\/developers.vtex.com\/docs\/guides\/authentication-overview#machine-authentication).",
```

Comment:
> To make this request, you need the order's changeRequestId, its unique change identification code. When you retry to change an order using its changeRequestId, instead of creating a new change, you avoid duplications.

This part would be greatly improved if you could add this info:
- Where do you get this ID? What request? What response status code? Probably 4xx or 5xx something, but it really depends on how the developers made it.
- What is the key under which you get this ID? Many APIs just return an `id` field or something like that instead of a more precise name. 

Also, it is not the "the order's `changeRequestId`" because one order can have many changes. It would be "the change's `changeRequestId`", or rather just "the `changeRequestId`" to avoid repetition.

Reply:
> Where do you get this ID? What request? What response status code? Probably 4xx or 5xx something, but it really depends on how the developers made it.

I am saying this because I imagine not all errors allow retries. An obvious example is something like `401 unauthorized`, but there are probably many others. So I am guessing there is a way to tell if you can retry or not.

Reply:
Vou adicionar os links do endpoint depois, mas melhorei essa questão de avisar onde se consegue o `changeRequestId`. Já a questão mais profunda dos erros vou investigar, mas não vou parar o PR por isso. Coloquei no DOC que estou usando para a organizar a task.  

Na época, eu perguntei para o time sobre erros, motivos, mas eles só disseram que seriam erros no geral, qlq um, daí ficou por isso.